artifact/v3/header/header.hpp

```diff
@@ -0,0 +1,178 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#ifndef MENDER_ARTIFACT_V3_HEADER_PARSER_HPP
+#define MENDER_ARTIFACT_V3_HEADER_PARSER_HPP
+
+#include <string>
+#include <utility>
+#include <vector>
+#include <unordered_map>
+
+#include <nonstd/optional.hpp>
+
+#include <common/expected.hpp>
+#include <common/error.hpp>
+#include <common/io.hpp>
+
+namespace mender {
+namespace artifact {
+namespace v3 {
+namespace header {
+
+using namespace std;
+
+using nonstd::nullopt;
+using nonstd::optional;
+
+namespace expected = mender::common::expected;
+namespace io = mender::common::io;
+namespace error = mender::common::error;
+
+//
+// +---header-info
+//
+
+enum class Payload {
+	RootfsImage,
```

Comment:
Do we actually need `RootfsImage`?

In the new implementation there shouldn't be any difference between `ModuleImage` and `RootfsImage`, because the details are handled by the Update Module.

Reply:
Very good point!

Reply:
I decided to keep it in here, due to the fact that we need to workaround that `type` is not written in `type-info:payload_type`